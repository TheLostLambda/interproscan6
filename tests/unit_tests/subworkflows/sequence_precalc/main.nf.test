nextflow_workflow {

    name "Test Workflow SEQUENCE_PRECALC"
    script "interproscan/subworkflows/sequence_precalc/main.nf"
    workflow "SEQUENCE_PRECALC"

    stage {
        symlink "interproscan/modules/lookup/check/main.nf"
        symlink "interproscan/modules/lookup/matches/main.nf"
        symlink "interproscan/modules/lookup/no_matches/main.nf"
        symlink "interproscan/scripts/lookup/lookup_check.py"
        symlink "interproscan/scripts/lookup/lookup_matches.py"
        copy "interproscan/scripts/lookup/lookup_no_matches.py"
        copy "tests/unit_tests/test_outputs/precalc_match_lookup/lookup_check_out"
        copy "tests/unit_tests/test_outputs/precalc_match_lookup/lookup_matches_out"
    }

    test("Run precalc match lookup successfully") {

        when {
            params {
                url_precalc = "https://www.ebi.ac.uk/interpro/match-lookup"
                check_precalc = "/isPrecalculated"
                matches = "/matches"
                sites = "/sites"
            }
            workflow {
                """
                input[0] = "tests/unit_tests/test_inputs/precalc_match_lookup/parsed_sequences"
                input[1] = "antifam,ncbifam"
                input[2] = true
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
