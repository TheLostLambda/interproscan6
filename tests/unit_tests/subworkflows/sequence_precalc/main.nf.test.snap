{
    "Run precalc match lookup successfully": {
        "content": [
            {
                "0": [
                    "no_match_lookup_fasta.fasta:md5,4e72b2094c713d10d64b89dbf482b483"
                ],
                "1": [
                    "parsed_match_lookup:md5,1291cc5ca7ba91ae1a6890f0735eec7e"
                ],
                "parsed_matches": [
                    "parsed_match_lookup:md5,1291cc5ca7ba91ae1a6890f0735eec7e"
                ],
                "sequences_to_analyse": [
                    "no_match_lookup_fasta.fasta:md5,4e72b2094c713d10d64b89dbf482b483"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-03T08:59:42.386748"
    }
}